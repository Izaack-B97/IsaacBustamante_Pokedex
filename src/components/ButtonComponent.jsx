import React from 'react'

export const ButtonComponent = ({ text = '' ,style = {}, handleClick = () => {} }) => {

  return (
    <div>
        <button className='btn' type="button" style={ style } onClick={ handleClick } >{ text }</button>
    </div>
  )
}
