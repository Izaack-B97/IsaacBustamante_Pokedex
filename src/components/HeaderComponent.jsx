import React from 'react'

export const HeaderComponent = () => {
  return (
    <>
        <h1>Listado de Pokemon</h1>
    </>
  )
}
