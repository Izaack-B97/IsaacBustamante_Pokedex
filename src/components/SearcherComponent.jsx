import React from 'react'

export const SearcherComponent = ({ handleFocus = () => {}, onQueryChanged = () => {} }) => {

  return (
    <div id='SearcherComponent'>
        <div className='search-container'>
            <img src={require('../assets/search-svgrepo-com.svg').default} alt='mySvgImage' />
            <input name='search-input' placeholder='Buscar' onChange={ onQueryChanged } onFocus={ handleFocus } />
        </div>
    </div>
  )
}
