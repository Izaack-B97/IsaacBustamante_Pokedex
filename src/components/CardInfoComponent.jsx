import React, { useEffect, useState } from 'react'
import { usePokemon } from '../hooks/usePokemon'
import { ButtonComponent } from './ButtonComponent'

export const CardInfoComponent = ( { pokemon, siguientesPokemos = () => {} } ) => {
  // console.log( pokemon )
  
  const [ sprites, setSprites ] = useState( [] );
  const [ moreDetails, setMoreDetails ] = useState( {} );
  
  const { pokemon : pokemonDetails, isLoading } = usePokemon( pokemon ? pokemon.id : {} );

  const mapSprites = () => {
    const sprites = pokemonDetails.sprites;
    const spritesArr = [];
    // console.log( sprites );
    for ( let key in sprites ) {
      if ( typeof sprites[ key ] === 'string' ) {
        spritesArr.push( sprites[ key ] );
      }      
    }

    setSprites( spritesArr );
  }
  
  const mapMoreDetails = () => {
    const types = pokemonDetails.types.map( ({ type }) => type.name);
    const moves = pokemonDetails.moves.map(({ move }) => move.name);

    setMoreDetails({ types, moves });
  }

  useEffect(() => {
    if ( Object.keys( pokemonDetails ).length > 0 ) {
      
      console.log( pokemonDetails )

      mapSprites();
      mapMoreDetails();

    }
  }, [ pokemonDetails ]);

  return (
    <div className='box2'>
      <div className='card-pokemon-info'>
        <div className='header-card-info'>
          <div className='card-pokemon-info-container'>
            { pokemon && <img src={ pokemon.picture } alt={ pokemon.name } /> }
          </div>
          { pokemon && <>
              <p># { pokemon.id }  </p>
              <p id='parrafo-name'>{ pokemon.name } </p>
            </> 
          }
        </div>
        <div className='details-pokemon'>
          <div>
            <h5>Types</h5>
            { Object.keys( moreDetails ).length > 0 && <span>{ moreDetails['types'].join(', ') }</span> }
          </div>
          <div>
            <h5>Peso</h5>
            <span>{ pokemonDetails.weight }kg</span>
          </div>
          <div>
            <h5>Sprites</h5>
            <div className='sprites-container'>
              { sprites.map(url => <img  key={url} src={ url } alt={ url } />) }
            </div>
          </div>
          <div>
            <h5>Movimientos</h5>
            { Object.keys( moreDetails ).length > 0 && <span>{ moreDetails['moves'].join(', ') }</span> }
          </div>
        </div>
      </div>
      <div className='buttons-container2'>
        <ButtonComponent text='Siguiente >' style={{ marginTop: '5px' }} handleClick={ siguientesPokemos } />
      </div>
    </div>
  )
}
