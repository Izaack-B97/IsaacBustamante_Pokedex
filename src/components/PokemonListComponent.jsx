import React from 'react'

import { ButtonComponent } from './ButtonComponent'

export const PokemonListComponent = ({ onClickCard = () => {}, simplePokemonList = [], anterioresPokemons = () => {} }) => {

  return (
    <div>
        <div className='list-container'>
        {
          simplePokemonList.map( pokemon => (
            <div key={ pokemon.name } className='card' onClick={ () => onClickCard( pokemon ) }>
              <div className='img-container'>
                <img src={ pokemon.picture } alt={ pokemon.name }/>
              </div>
              <h5># { pokemon.id }</h5>
              <p># { pokemon.name }</p>
            </div>         
          ))
        }
        </div>
        <div className='buttons-container'>
            <ButtonComponent text='< Atras' handleClick={ anterioresPokemons } style={{ marginLeft: '10px' }}/>
        </div>
    </div>
  )
}
