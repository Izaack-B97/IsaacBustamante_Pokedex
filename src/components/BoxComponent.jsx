import React from 'react'

import { PokemonListComponent } from './PokemonListComponent'
import { SearcherComponent } from './SearcherComponent'

export const BoxComponent = ({ 
  onClickCard = () => {}, 
  simplePokemonsList = [], 
  anterioresPokemons = () => {} ,
  handleFocus = () => {},
  onQueryChanged = ()=> {}
}) => {

  return (
    <div className='box1'>
        <SearcherComponent handleFocus={ handleFocus } onQueryChanged={ onQueryChanged } />
        <PokemonListComponent  onClickCard={onClickCard} simplePokemonList={ simplePokemonsList } anterioresPokemons={ anterioresPokemons }/>
    </div>
  )
}
