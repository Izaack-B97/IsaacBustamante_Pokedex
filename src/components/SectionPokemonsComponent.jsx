import React, { useRef, useState } from 'react'

import { BoxComponent } from './BoxComponent'
import { CardInfoComponent } from './CardInfoComponent'
import { usePokemonPaginated } from '../hooks/usePokemonPaginated';

export const SectionPokemonsComponent = () => {
  
  const [ pokemonSelected, setPokemonSelected ] = useState(null);
  
  const { simplePokemonList, loadPokemons, setSimplePokemonList } = usePokemonPaginated();

  const debounceRef = useRef();

  const onClickCard =  ( pokemon ) => {
    // console.log( pokemon );
    setPokemonSelected( pokemon );
  }

  const siguientesPokemons = () => {
    loadPokemons();
  }

  const anterioresPokemons = () => {
    loadPokemons( true );
  }

  const onQueryChanged = ( event ) => {
    if ( debounceRef.current ) clearTimeout(  debounceRef.current );

    debounceRef.current = setTimeout(() => {

      if ( event.target.value.length > 0 ) {
        console.log( 'debounced value:', event.target.value );

        const newArr = simplePokemonList.filter( pokemon => pokemon.name.includes( event.target.value ) );
        console.log( newArr );
        setSimplePokemonList( newArr );
      } else {
        loadPokemons( false, true );
      }
    }, 350);
  }

  const handleFocus = () => {
    // loadPokemons( false, 1154); // All Pokemxons
    // console.log( simplePokemonList );

  }

  return (
    <div>
      <div>
        <div className='section'>
          <BoxComponent
            onClickCard={ onClickCard }  
            simplePokemonsList={ simplePokemonList } 
            anterioresPokemons={ anterioresPokemons }
            handleFocus={ handleFocus }
            onQueryChanged={ onQueryChanged }
          />
          <CardInfoComponent pokemon={ pokemonSelected } siguientesPokemos={ siguientesPokemons } />
        </div>
      {/* Buttons Section */}
        {/* <div className='buttons-container'>
            <ButtonComponent style={{ 'margin-left': '100px' }} />
            
        </div> */}
      </div>
    </div>
  )
}
