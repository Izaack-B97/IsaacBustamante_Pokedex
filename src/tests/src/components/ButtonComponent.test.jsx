import { render, screen } from "@testing-library/react";
import { ButtonComponent } from "../../../components/ButtonComponent";

describe('Pruebas en <ButtonComponent />', () => {
    
    it('Debe hacer match con el snapshot', () => {
        const { container } = render( <ButtonComponent text="Click me" /> )
        
        expect( container ).toMatchSnapshot();
        // expect( screen.getByText( /Click me/i ) ).toBeInTheDocument();
    })

});