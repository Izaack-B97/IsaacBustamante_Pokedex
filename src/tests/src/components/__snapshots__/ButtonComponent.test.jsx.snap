// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ButtonComponent /> Debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <button
      class="btn"
      type="button"
    >
      Click me
    </button>
  </div>
</div>
`;
